hydromt/data_adapter/rasterdataset.py

```diff
@@ -271,69 +282,81 @@ def to_file(
+        cache_root: Optional[StrPath] = None,
+        logger: Logger = logger,
     ):
         """Return a clipped, sliced and unified RasterDataset.
 
         For a detailed description see:
         :py:func:`~hydromt.data_catalog.DataCatalog.get_rasterdataset`
         """
-        # load data
-        fns = self._resolve_paths(time_tuple, variables, zoom_level, geom, bbox, logger)
-        self.mark_as_used()  # mark used
-        ds = self._read_data(
-            fns,
-            geom,
-            bbox,
-            cache_root,
-            handle_nodata=handle_nodata,
-            zoom_level=zoom_level,
-            logger=logger,
-        )
-        # rename variables and parse data and attrs
-        ds = self._rename_vars(ds)
-        ds = self._validate_spatial_dims(ds)
-        ds = self._set_crs(ds, logger)
-        ds = self._set_nodata(ds)
-        ds = self._shift_time(ds, logger)
-        # slice data
-        ds = RasterDatasetAdapter._slice_data(
-            ds,
-            variables,
-            geom,
-            bbox,
-            buffer,
-            align,
-            time_tuple,
-            handle_nodata=handle_nodata,
-            logger=logger,
-        )
-        # uniformize data
-        ds = self._apply_unit_conversions(ds, logger)
-        ds = self._set_metadata(ds)
-        # return array if single var and single_var_as_array
-        return self._single_var_as_array(ds, single_var_as_array, variables)
+        try:
+            # load data
+            fns = self._resolve_paths(
+                time_tuple, variables, zoom_level, geom, bbox, logger
+            )
+            ds = self._read_data(
+                fns,
+                geom,
+                bbox,
+                cache_root,
+                zoom_level=zoom_level,
+                logger=logger,
+            )
+            if has_no_data(ds):
+                raise NoDataException()
+            # rename variables and parse data and attrs
+            ds = self._rename_vars(ds)
+            ds = self._validate_spatial_dims(ds)
+            ds = self._set_crs(ds, logger)
+            ds = self._set_nodata(ds)
+            ds = self._shift_time(ds, logger)
+            # slice data
+            ds = RasterDatasetAdapter._slice_data(
+                ds,
+                variables,
+                geom,
+                bbox,
+                buffer,
+                align,
+                time_tuple,
+                logger=logger,
+            )
+            if has_no_data(ds):
+                raise NoDataException()
+
+            # uniformize data
+            ds = self._apply_unit_conversions(ds, logger)
+            ds = self._set_metadata(ds)
+            self.mark_as_used()  # mark used
+            # return array if single var and single_var_as_array
+            return self._single_var_as_array(ds, single_var_as_array, variables)
+        except (NoDataException, FileNotFoundError):
```

Comment:
I passed it for other adapters but actually I think it is best to still have two separate error messages for NoDataExecption and FileNotFoundError.
If FileNotFoundError, it could be that the user made a mistake in the path in the data catalog, so I still think it's good to return an error message that let's them know the file was not found rather than just ignore in that case.